bandits/ordinarylinearbandit.py

```diff
@@ -0,0 +1,93 @@
+from absl import logging
+import numpy as np
+from arms import BernoulliArm
+from arms import LinearArm
+from .utils import Bandit
+
+__all__ = ['OrdinaryLinearBandit']
+
+class OrdinaryLinearBandit(Bandit):
+  """Ordinary linear bandit model
+  Arms are numbered from 0 to len(arms)-1 by default.
+  """
+
+  def __init__(self, arms, theta):
+    logging.info('Ordinary linear bandit model')
+    if not isinstance(arms, list):
+      logging.fatal('Arms should be given in a list!')
+    for arm in arms:
+      if not isinstance(arm, LinearArm):
+        logging.fatal('Not a linear arm!')
+
+    self.__theta = np.array(theta)
+    self.__arms = arms
+
+    for idx, arm in enumerate(self.__arms):
+      if arm.action.shape !=self.theta.shape:
+        logging.fatal('The action and global parameter dimensions are unequal!')
+
+    self.__arm_num = len(arms)
+    if self.__arm_num < 2:
+      logging.fatal('The number of arms should be at least two!')
+
+    self.__best_arm_ind = max([(tup[0], np.dot(tup[1].action,self.theta))
+        for tup in enumerate(self.__arms)], key=lambda x:x[1])[0]
+    self.__best_arm = self.__arms[self.__best_arm_ind]
+
+  @property
+  def arm_num(self):
+    """return number of arms"""
+    return self.__arm_num
+
+  @property
+  def arms(self):
```

Comment:
Should this property be protected?

Reply:
So one question that I had: for this correlated/linear bandits, each action corresponds to a vector (defined in the config under 'means' key) that is known to the player.  I unprotected this arm property so I could initialize this vector for em_arms in cucb.py line 27.
```
for k in range(self._arm_num):
      self._em_arms[k].action = self._bandit.arms[k].action
```
What would be the correct way that you recommend to pass this action vector that is property of the arm to the empirical arm so the learner can reference it?

Reply:
At this point, I suggest putting a property under `bandit` class and let the learner communicate only with the bandit. Also, can you pull my new commit? In my new commit, I update the config file to the following style and arm setup is now put under the bandit initialization. And then you can define a property telling the learners that what actions are.

```
{
	"environment": {
                "type": "OrdinaryCorrelatedBandit",
		"OrdinaryCorrelatedBandit": {	
			"type": "CorrelatedArm",
			"means": [[0.3,0.2], [0.5,0.6], [0.7,0.9]],
                        "param": [0.5,0.5]
		}
	},
	"learner": {
		"goal": "regretmin",
		"type": "ordinarylearner",
		"policy": ["Uniform","CUCB","UCB"]
	},
	"parameters": {
		"horizon": 2000,
		"freq": 20,
		"trials": 200,
		"processors": 40
	}
}
```

Reply:
Commit a refactor for compatibility w/ the config file & added actions property under CorrelatedBandit class, separated correlatedlearners.